### PR TITLE
Improved "Open URL in Simulator" command (raycast/extensions#4533)

### PR DESCRIPTION
**Pull request by SvenTiigi**
## Changes

- Refactored implementation of `XcodeSimulatorService.openUrl` function by removing usage of Raycast APIs which allows a more consistent invocation across other components.
- Improved "Open URL in Simulator" command by checking the reason why a url couldn't be opened and redirecting the end user to the "Manage Simulators" command if necessary.
- Changed the URL argument `required` property value of the "Open URL in Simulator" command to `true` as the previous implementation was using the Clipboard as a fallback which might not be 100% clear to the end user so it would be better to make it a required argument.
- Added a dedicated Open URL Form (`xcode-simulator-open-url-form.component`) to the "Manage Simulators" command which prefills the textfield with the contents of the clipboard if it is a valid url to be opened in a simulator.
- Removed the search bar accessory in the "Manage Simulators" command as I think it would be better to add this functionality in a separate PR to allow filtering either by the state of the simulator or the runtime e.g. iOS 16, tvOS 16, etc.

### CODE REVIEW

File: extensions/xcode/src/open-url-in-simulator.command.tsx [hunk 1]
```diff
@@ -1,10 +1,53 @@
...
+          toast.style = Toast.Style.Failure;
+          toast.title = "Please enter a valid url";
+          break;
+        case XcodeSimulatorOpenUrlErrorReason.bootedSimulatorMissing:
+          // Hide toast
+          await toast.hide();
+          // Show info hud
+          await showHUD("Please boot a simulator to open a url");
+          // Launch manage simulators command in order to boot a simulator
+          await launchCommand({ name: "manage-simulators.command", type: LaunchType.UserInitiated });
```
Comment (qeude) [inline extensions/xcode/src/open-url-in-simulator.command.tsx]: I find it to be possibly weird here. By doing so, on UX perspective we could expect that the url opens right after the selected simulator finished booting. That's why I prefered a classic error message. WDYT?

File: extensions/xcode/src/open-url-in-simulator.command.tsx [hunk 2]
```diff
@@ -1,10 +1,53 @@
...
+    // Try to open url
+    await XcodeSimulatorService.openUrl(props.arguments.url);
+    // Close main window
+    await closeMainWindow();
+  } catch (error) {
+    // Check if error is an instance of XcodeSimulatorOpenUrlError
+    if (error instanceof XcodeSimulatorOpenUrlError) {
+      // Switch on reason
+      switch (error.reason) {
+        case XcodeSimulatorOpenUrlErrorReason.badUrl:
```
Comment (qeude) [inline extensions/xcode/src/open-url-in-simulator.command.tsx:24]: Way better error handling here, thanks 🙏

File: extensions/xcode/package.json [hunk 3]
```diff
@@ -251,7 +251,7 @@
           "name": "url",
           "placeholder": "URL to open",
```
Comment (qeude) [inline extensions/xcode/package.json]: We might want to change this placeholder to an example such as in `XcodeSimulatorOpenUrlForm`, WDYT?

File: extensions/xcode/src/components/xcode-simulator-list/xcode-simulator-open-url-form.component.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm
+            title="Open URL"
+            onSubmit={() =>
+              operationWithUserFeedback(
+                "Opening URL in Simulator",
+                "URL successfully opened in simulator",
+                "An error occurred while trying to open the url",
+                () => XcodeSimulatorService.openUrl(url, props.simulator)
```
Comment (qeude) [inline extensions/xcode/src/components/xcode-simulator-list/xcode-simulator-open-url-form.component.tsx:23]: Couldn't we use the same error management as in the command so it's the same anywhere we use it?

**Review by qeude — COMMENTED** [review]
Thanks for the PR, very interesting to learn from your solution here. 
I have some questions and suggestions though before approving 🙏

File: extensions/xcode/src/open-url-in-simulator.command.tsx (hunk 1 above)
Comment (SvenTiigi) [reply extensions/xcode/src/open-url-in-simulator.command.tsx]: I have refactored the implementation to a failure toast with a primary action allowing to easily open the "Manage Simulators" command 👌 (https://github.com/qeude/extensions/pull/1/commits/33909b321812a402bbf4800bbae7c62742f75db2)

Bildschirm­foto 2023-02-05 um 16 16 40

File: extensions/xcode/package.json (hunk 3 above)
Comment (SvenTiigi) [reply extensions/xcode/package.json]: Updated with https://github.com/qeude/extensions/pull/1/commits/c2de4ab286c5a9a60daa394327099a8bdf696ec9

File: extensions/xcode/src/components/xcode-simulator-list/xcode-simulator-open-url-form.component.tsx (hunk 4 above)
Comment (SvenTiigi) [reply extensions/xcode/src/components/xcode-simulator-list/xcode-simulator-open-url-form.component.tsx:23]: As the context is a little bit different in the `XcodeSimulatorOpenUrlForm` component we can simply ignore some errors for example:

- Xcode installation is guaranteed to be available as the `XcodeSimulatorList` is wrapped in a `XcodeInstallationVerifier` component.
- Missing booted simulator is not possible as the `Action.Push` to open the form is only available on a booted simulator.

Therefore wrapping the `XcodeSimulatorService.openUrl` in a `operationWithUserFeedback` should do the job.

**Review by qeude — APPROVED** [review]
LGTM, thanks for these changes 🚀

**Pull request merged**